hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobRepository.java

```diff
@@ -458,6 +481,13 @@ public JobExecutionRecord getJobExecutionRecord(long jobId) {
         return Util.memoizeConcurrent(() -> instance.getMap(RESOURCES_MAP_NAME_PREFIX + idToString(jobId)));
     }
 
+    /**
+     * Gets the job files storage map, lazily evaluated to avoid creating the map if it won't be needed
+     */
+    Supplier<IMap<String, byte[]>> getJobFileStorage(long jobId) {
+        return Util.memoizeConcurrent(() -> instance.getMap(jobFileStorageMapName(jobId)));
```

Comment:
We should make it lazy at caller, not here. Here we can get the map unconditionally.